bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationConfigurationHandler.java

```diff
@@ -0,0 +1,163 @@
+                loginPostStringBuilder.append(Integer.toString((mDigestedPassword[i] & 0xff) + 0x100, 16).substring(1));
+            }
+            String saltedmDigestedPwd = Base64.getEncoder().encodeToString(mDigest.digest(saltedPassword.getBytes()));
+
+            String loginPostJsonData = "{\"mode\":1,\"userId\":\"" + username + "\",\"pwh\":\"" + saltedmDigestedPwd
+                    + "\"}";
+
+            Request loginPostJsonResponse = httpClient.POST(urlLogin + "?sessionId=" + sessionId);
+            loginPostJsonResponse.header(HttpHeader.CONTENT_TYPE, "application/json");
+            loginPostJsonResponse.content(new StringContentProvider(loginPostJsonData));
+            ContentResponse loginPostJsonDataContentResponse = loginPostJsonResponse.send();
+
+            String loginPostResponse = new String(loginPostJsonDataContentResponse.getContent(),
+                    StandardCharsets.UTF_8);
+
+            JsonObject loginPostJsonObject = (JsonObject) new JsonParser()
+                    .parse(transformJsonResponse(loginPostResponse));
+
+            sessionId = extractSessionId(loginPostJsonObject);
+
+            // Part for sending data to Inverter
+            String postJsonData = "{\"dxsEntries\":[{\"dxsId\":" + dxsId + ",\"value\":" + value + "}]}";
+
+            Request postJsonDataRequest = httpClient.POST(url + "/api/dxs.json?sessionId=" + sessionId);
+            postJsonDataRequest.header(HttpHeader.CONTENT_TYPE, "application/json");
+            postJsonDataRequest.content(new StringContentProvider(postJsonData));
+            postJsonDataRequest.send();
+        } catch (JsonIOException getAuthenticateResponseException) {
+            logger.debug("Could not read the response: {}", getAuthenticateResponseException.getMessage());
+        }
+    }
+
+    static String transformJsonResponse(String jsonResponse) {
+        // Method transformJsonResponse converts response,due to missing [] in ContentResponse
+        // postJsonDataContentResponse.
+
+        int sessionStartPosition = jsonResponse.indexOf("session");
+        int statusStartPosition = jsonResponse.indexOf("status");
+
+        StringBuilder transformStringBuilder = new StringBuilder();
+
+        transformStringBuilder.append(jsonResponse);
+
+        transformStringBuilder.insert(sessionStartPosition + 9, '[');
+        int roleIdStartPosition = jsonResponse.indexOf("roleId");
+        transformStringBuilder.insert(roleIdStartPosition + 11, ']');
+
+        transformStringBuilder.insert(statusStartPosition + 10, '[');
+        int codeStartPosition = jsonResponse.indexOf("code");
+        transformStringBuilder.insert(codeStartPosition + 11, ']');
+
+        String transformJsonObject = transformStringBuilder.toString();
+
+        return transformJsonObject;
+    }
+
+    // Method extractSessionId extracts sessionId from JsonObject
+    static String extractSessionId(JsonObject extractJsonObjectSessionId) throws TimeoutException {
```

Comment:
This exception is never thrown and can therefore be removed:
```suggestion
    static String extractSessionId(JsonObject extractJsonObjectSessionId) {
```